### PR TITLE
Bloyal tensorflow iris byom

### PR DESCRIPTION
**Pull request by brianloyal**
*Issue #, if available:*

*Description of changes:*
IHAC who is looking for a good example of creating a SageMaker endpoint from a pre-trained TensorFlow model. Seems like this notebook does the trick, but it was created using TF 1.x. Since there have been some significant changes in TF 2.x, I updated the notebook to use the newer version. 

*Testing done:*
"Run All" completed successfully on a ml.g4dn.xlarge running the TF 2.3/Py3.7 GPU kernel.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-sagemaker-examples/blob/master/CONTRIBUTING.md) doc and adhered to the example notebook best practices
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-sagemaker-examples/blob/master/README.md)
- [x] I have tested my notebook(s) and ensured it runs end-to-end
- [x] I have linted my notebook(s) and code using `tox -e black-format,black-nb-format`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: advanced_functionality/tensorflow_iris_byom/tensorflow_BYOM_iris.ipynb [hunk 1]
```diff
@@ -22,6 +22,13 @@
     "__Note__: Compare this with the [tensorflow bring your own model example](../tensorflow_iris_byom/tensorflow_BYOM_iris.ipynb)"
```
Comment (jkroll-aws) [inline advanced_functionality/tensorflow_iris_byom/tensorflow_BYOM_iris.ipynb:22]: <p>What is the reason for switching from a separate .py file to having the code in the notebook? Just being more cohesive, or anything else? Since it's only ~30 lines of code, I don't mind having it directly in the notebook like this.</p><p> </p><p>Related, why is an <code>entry_point</code> no longer needed?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3100/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (jkroll-aws) [inline advanced_functionality/tensorflow_iris_byom/tensorflow_BYOM_iris.ipynb:22]: <pre class="ql-syntax" spellcheck="false">Line #2.    IRIS_TRAINING_URL = "http://download.tensorflow.org/data/iris_training.csv"
</pre><p>While you're revising this notebook, can you please switch from the TensorFlow hosted version of Iris to our AWS-hosted version? S3 paths are:</p><pre class="ql-syntax" spellcheck="false">s3://sagemaker-sample-files/datasets/tabular/iris/iris_train.csv
s3://sagemaker-sample-files/datasets/tabular/iris/iris_test.csv
</pre><p>Then rather than using the <code>request</code>  library, you can use boto3.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3100/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (jkroll-aws) [inline advanced_functionality/tensorflow_iris_byom/tensorflow_BYOM_iris.ipynb:22]: <pre class="ql-syntax" spellcheck="false">Line #1.    EPOCHS = 50
</pre><p>What is the approximate runtime with 50 epochs? If it's more than ~1 hour, please lower the number of epochs and add a note that the customer can increase for better accuracy.</p><p> </p><p>I see you're also removing lines of explanatory text. As you revise the code, add new explanations describing what this revised version does. The examples should educate the customer as well as providing them with working code.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3100/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

**Comment by jkroll-aws** [comment]
Thanks for the PR! Great to update to TF 2.x.

File: advanced_functionality/tensorflow_iris_byom/tensorflow_BYOM_iris.ipynb (hunk 1 above)
Comment (brianloyal) [reply advanced_functionality/tensorflow_iris_byom/tensorflow_BYOM_iris.ipynb:22]: Good idea!

Comment (brianloyal) [reply advanced_functionality/tensorflow_iris_byom/tensorflow_BYOM_iris.ipynb:22]: Because the model is so small, I think a lot of folks would just define it in a notebook cell, especially since we need to load it for on-notebook training anyhow. If it were larger, or required additional processing tasks, I'd probably put it in it's own file.

As for the entry_point, as I understand it when you run `.save` on a Tensorflow model it includes all the code necessary to run it in the `saved_model.pb` file.

https://www.tensorflow.org/guide/saved_model

So, the TensorFlow estimator must have some logic to extract that automagically and use it as the inference fn. ¯\_(ツ)_/¯

Comment (brianloyal) [reply advanced_functionality/tensorflow_iris_byom/tensorflow_BYOM_iris.ipynb:22]: It trains in about 5 seconds - the benefits of using a super-simple model definition. I've added a note to the markdown.

**Comment by brianloyal** [comment]
Updated based on feedback from @jkroll-aws. Thanks!

**Review by atqy — DISMISSED** [review]

**Review by atqy — APPROVED** [review]
